verification/src/transaction_verifier.rs

```diff
@@ -225,14 +225,15 @@ impl<'a> MaturityVerifier<'a> {
 
     pub fn verify(&self) -> Result<(), TransactionError> {
         let cellbase_immature = |meta: &CellMeta| -> bool {
-            meta.is_cellbase()
-                && self.block_number
-                    < meta
-                        .transaction_info
-                        .as_ref()
-                        .expect("cell meta should have block number when transaction verify")
-                        .block_number
-                        + self.cellbase_maturity
+            let cell_block_number = meta
+                .transaction_info
+                .as_ref()
+                .expect("cell meta should have block number when transaction verify")
```

Comment:
`TransactionInfo` is missing if the cell is from a transaction in the pool. I dislike the smell of the old code, we should not rely that transaction info must exist when `is_cellbase` returns true. Let's be more verbose that:

- Add a method `is_cellbase` in `TransactionInfo`.
- `CellMeta` can call `TransactionInfo::is_cellbase` in its `is_cellbase`
- Use `meta.transaction_info.map(|info| info.is_cellbase() && info.block_number > 0 && ...) = Some(true)` here.